src/Raven.Server/Documents/Sharding/Handlers/ShardedSubscriptionsHandler.cs

```diff
@@ -32,13 +32,25 @@ public async Task Update()
             using (var processor = new ShardedSubscriptionsHandlerProcessorForPostSubscription(this))
                 await processor.ExecuteAsync();
         }
-        
+
+        internal void ThrowOnRevisionsIfNeeded(bool subRevisions)
+        {
+            if (subRevisions)
+            {
+                // RavenDB-18881
+                throw new NotImplementedException(@"Revisions subscription is not supported for sharding database.");
+            }
+        }
+
         public async Task CreateSubscriptionInternalAsync(BlittableJsonReaderObject bjro, long? id, bool? disabled, SubscriptionCreationOptions options, JsonOperationContext context)
```

Comment:
This code looks like t should be inside a processor, not in handler?

Reply:
its used in both  `AbstractSubscriptionsHandlerProcessorForPostSubscription` & `AbstractSubscriptionsHandlerProcessorForPutSubscription` so I guesss thats why it was put here

Reply:
Move it to one of them, and use static from there, let's try to keep handlers clean.

Reply:
created new helper class